### PR TITLE
Sockperf filewrite

### PR DESCRIPTION
**Pull request by garg7pec**

### CODE REVIEW

File: src/server.cpp [hunk 1]
```diff
@@ -318,6 +321,22 @@ int Server<IoType, SwitchActivityInfo, SwitchCalcGaps>::server_accept(int ifd) {
                                         inet_ntoa(addr.sin_addr), ntohs(addr.sin_port), active_ifd);
                             }
                             do_accept = true;
+
+                            if (s_user_params.sockperfDumpDataToFile)
+                            {
+                                retVal = system((createSockperfDirectoryCommand.append(s_user_params.sockperfDumpDataDirectory)).c_str());
```
Comment (rafiw) [inline src/server.cpp:327]: Why not to create using the mkdir function?https://codeyarns.com/2014/08/07/how-to-create-directory-using-c-on-linux/

File: src/server.h [hunk 2]
```diff
@@ -150,6 +153,17 @@ void close_ifd(int fd, int ifd, fds_data *l_fds_ifd) {
             break;
         }
     }
+    if(bufferDumpFile)
+        fclose(bufferDumpFile);
+}
+//------------------------------------------------------------------------------
+
+std::string generateDumpFileName(std::string ip)
```
Comment (rafiw) [inline src/server.h:161]: Where is the IP used?

**Review by rafiw — CHANGES_REQUESTED** [review]

**Comment by rafiw** [comment]
Hi @garg7pec thank you for your contribution
What is the motivation of this change?

**Comment by garg7pec** [comment]
the motive for this change is to add capability in sockperf tool to do the disk writes for the files (in sync manner to be sure that the disk write has happened for sure).

**Comment by igor-ivanov** [comment]
Hi @garg7pec `sockperf` is considered as a tool to measure performance on network. It seems that motivation you pointed does not meet primary usage case.

**Comment by igor-ivanov** [comment]
rejected

**Pull request closed without merging**